src/useDebouncedRef/useDebouncedRef.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { afterEach, vi, expect, describe, it } from 'vitest'
+import { ref } from 'vue'
+import { useDebouncedRef } from '@/useDebouncedRef/useDebouncedRef'
+
+afterEach(() => {
+  vi.useRealTimers()
+})
+
+describe('useDebouncedRef', () => {
+
+  it('it syncs the initial value', () => {
+    const input = ref(1)
+    const output = useDebouncedRef(input, 100)
+
+    expect(output.value).toBe(1)
+  })
+
+  it('it debounces updates to the value', async () => {
+    vi.useFakeTimers()
+    const input = ref(1)
+    const output = useDebouncedRef(input, 100)
+    input.value = 2
+
+    await vi.runAllTimersAsync()
+
+    expect(output.value).toBe(2)
+  })
+
+  it('it syncs the debounced value back to the input', () => {
+    vi.useFakeTimers()
+    const input = ref(1)
+    const output = useDebouncedRef(input, 100)
+    output.value = 2
+
+    expect(input.value).toBe(2)
```

Comment:
how does this pass without `runAllTimers`?

Reply:
There's no debounce syncing from output to input. Only when setting input will there be a debounce in syncing that value to the output.

Reply:
ah, yeah I had that backwards in my mind.

Reply:
I wonder if the contract I envisioned is actually more straightforward?

Reply:
```
const myActualValue = ref(1)
const theDebouncedVersion = useDeboundedRef(myActualValue)

myActualValue.value = 2 // happens immediately
theDebouncedVersion.value = 2 // awaits the delay
```

Reply:
oh interesting. I think of the debounced ref as a copy that is synced using a debounce. The lines get a little blurred though when you can set either value. I could see it both ways. But since we're already using it the input -> debounce -> output way I think we should keep it as is.